src/pages/mypage/index.tsx

```diff
@@ -37,28 +47,76 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
     notFound: true,
   };
 };
-
+const style = {
+  position: 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+};
 type MyPageProps = {
   username: string;
   firstname: string;
   lastname: string;
   loggedIn: boolean;
+  profileImageCookie: string;
+  handleImageChange: () => void;
 };
 const MyPage: NextPage<MyPageProps> = ({
   username,
   firstname,
   lastname,
   loggedIn,
+  profileImageCookie,
 }) => {
   const { isLoggedIn, changeLogInState } = useContext(LoggedInContext);
+  const [profileImage, setProfileImage] = useState('');
+  const [open, setOpen] = React.useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  useEffect(() => {
+    console.log(profileImageCookie);
+    if (profileImageCookie) {
+      setProfileImage(profileImageCookie);
+    } else {
+      setProfileImage('');
+    }
+  }, []);
```

Comment:
it gives an error that it is missing the dependency because profileImageCookie can be changed, but useEffect wouldn't know about it. If your useEffect should be triggered when profileImageCookie then, add profileImageCookie in [ ],

Reply:
Ah okey. Goal was to set if(profileImageCookie) then setProfileImage to profileImageCookie when component renders for the first time only.

Reply:
> Ah okey. Goal was to set if(profileImageCookie) then setProfileImage to profileImageCookie when component renders for the first time only.

I just checked, seems like profileImageCookie is not changable so it will be okay to write it in useEffect [ ]